main.go

```diff
@@ -92,11 +93,18 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = (&controllers.AWSAdapterConfigReconciler{
-		Client:          mgr.GetClient(),
+	cl, err := client.New(ctrl.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		setupLog.Error(err, "unable to create client")
+		os.Exit(1)
+	}
+
+	r := &controllers.AWSAdapterConfigReconciler{
+		Client:          cl,
```

Comment:
Move client creation, check etc to an independent function below main.

Reply:
Done.
We do need reconciler to call IsPresent and Create methods